editor.js

```diff
@@ -529,32 +545,36 @@ $(document).ready( () => {
             // Call checkLeave only if we are NOT loading a new project
             if (getURLParameter('openFile') === "true") {
                 return;
-            } 
-            /*
-            // This code attempts to save the current workspace into the localStorage.
-            // Testing shows that when this is implemented, the checkLeave below is never reached,
-            // preventing the dialog warning the user that their current project is not saved.
-            else {
-                // store the current project into the localStore so that if the page is
-                // being refreshed, then it will automatically be reloaded
-                projectData.timestamp = this.performance.now();
-                var pd = projectData;
-                pd.code = getXml();
-                window.localStorage.setItem('localProject', JSON.stringify(pd));
             }
-            */ 
+
+            // ------------------------------------------------------
+            // This code attempts to save the current workspace into
+            // the localStorage. Testing shows that when this is
+            // implemented, the checkLeave below is never reached,
+            // preventing the dialog warning the user that their
+            // current project is not saved.
+            // Note:
+            //   the call to this.performance.now() to get the
+            //   timestamp was failing. Replacing that call with a
+            //   generic function getTimestamp() resolves the error.
+            // ------------------------------------------------------
+
+            // Store the current project into the localStore so that
+            // if the page is being refreshed, it will automatically
+            // be reloaded
+            // ------------------------------------------------------
+            let tempProject = {};
+            Object.assign(tempProject, projectData);
+
+            tempProject.code = getXml();
+            tempProject.timestamp = getTimestamp();
+            window.localStorage.setItem('localProject', JSON.stringify(tempProject));
         }
 
         if (checkLeave()) {
             e.preventDefault();     // Cancel the event
             e.returnValue = Blockly.Msg.DIALOG_CHANGED_SINCE;
             return Blockly.Msg.DIALOG_CHANGED_SINCE;
-        } else if (isOffline) {
```

Comment:
This block is removed now that the calls to this.performance.now() have been removed.